src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -319,7 +358,16 @@ private void getNewBrapiData(List<BrAPIImport> importRows, List<Column<?>> pheno
             this.observationUnitByNameNoScope.put(key, obsUnitPIO);
 
             for (Column<?> column : phenotypeCols) {
-                PendingImportObject<BrAPIObservation> obsPIO = createObservationPIO(importRow, column.name(), column.getString(i));
+                //If associated timestamp column, add
+                String dateTimeValue = null;
+                if (timeStampColByPheno.get(column.name())!=null) {
+                    dateTimeValue = timeStampColByPheno.get(column.name()).getString(i);
+                    //If no timestamp, set to midnight
+                    if (!validDateTimeValue(dateTimeValue) && !dateTimeValue.isBlank()){
```

Comment:
Your logic here is sound.  Minor suggestion, switch the order of the checks so that the `isBlank` check is first.  This is easier to compute than validating the date/time, and can reduce the amount of work the system has to do when importing a file

Reply:
Good point, switched